lib/base/Domain.js

```diff
@@ -21,7 +21,20 @@ function Domain(twilio, baseUrl) {
  * @return {string} absolute url
  */
 Domain.prototype.absoluteUrl = function(uri) {
-  return _.trim(this.baseUrl, '/') + '/' + _.trim(uri, '/');
+  var result = '';
+  if (typeof this.baseUrl === 'string') {
+    const cleanBaseUrl = trim(this.baseUrl, '/');
+    result += cleanBaseUrl;
+    result += '/';
+  }
+  if (typeof uri === 'string') {
+    uri = trim(uri, '/');
+    if (result === '') {
+      result += '/';
+    }
+    result += uri;
+  }
+  return result;
```

Comment:
Could make this into a `join` function or something that removes the leading/trailing delim from each and then joins them using the delim if both are non-empty strings.